File: java/leetcode_grind/Day715.java
package leetcode_grind;

public class Day715 {
    // https://leetcode.com/problems/delete-characters-to-make-fancy-string/description/?envType=daily-question&envId=2024-11-01
    static class Solution1 {
        public String makeFancyString(String s) {
            var ans = new StringBuilder();
            ans.append(s.charAt(0));
            for (int i = 1, cnt = 1; i < s.length(); i++) {
                if (s.charAt(i) == ans.charAt(ans.length() - 1)) {
                    cnt++;
                } else {
                    cnt = 1;
                }
                if (cnt < 3) {
                    ans.append(s.charAt(i));
                }
            }
            return ans.toString();
        }
    }
}

File: src/leetcode_grind/day_715.rs
// https://leetcode.com/problems/delete-characters-to-make-fancy-string/description/?envType=daily-question&envId=2024-11-01
pub fn make_fancy_string(s: String) -> String {
    let s = s.chars().collect::<Vec<_>>();
    let mut ans = String::new();
    ans.push(s[0]);
    if s.len() > 1 {
        ans.push(s[1]);
    }
    for i in 2..s.len() {
        if s[i] == s[i - 1] && s[i - 1] == s[i - 2] {
            continue;
        }
        ans.push(s[i] as char);
    }
    ans
}
